classes/WP_CLI/Migration_Command.php

```diff
@@ -75,6 +75,16 @@ public function register() {
 	 * @return void
 	 */
 	public function migrate( $positional_args, $assoc_args ) {
+		if ( ! Controller::instance()->allow_migration() ) {
+			WP_CLI::error( 'Migration prevented due to missing depdendencies or by current data store.' );
+			return;
+		}
+
+		if ( \ActionScheduler_DataController::is_migration_complete() ) {
```

Comment:
@crstauf Could you move this to be the first condition check? When writing early exit checks it is preferable that the most common conditions be tested first.

Reply:
@rrennick Done.